react-1/src/components/User.js

```diff
@@ -0,0 +1,119 @@
+import React, {useImperativeHandle, useRef} from 'react';
+import { DragSource, DropTarget } from 'react-dnd';
+import {createUseStyles} from 'react-jss';
+import { useSpring, animated } from 'react-spring';
+import ItemTypes from './ItemTypes';
+
+const useStyles = createUseStyles({
+    User: {
+        minHeight: 40,
+        padding: 40,
+        backgroundColor: '#fff',
+        color: '#949494',
+        cursor: 'move'
+    },
+	Container: {
+		borderBottom: '1px solid #e4e4e4',
+		"&:last-child": {
+			borderBottom: 'inherit'
+		},
+	},
+    Infos: {float: 'left'},
+    UserName: {
+        fontWeight: 'bold',
+        fontSize: 20,
+        float: 'left',
+        marginBottom: 5
+    },
+    UserEmail: {
+        textDecoration: 'none',
+        color: '#949494',
+        float: 'left'
+    },
+    Arrow: {float: 'right'}
+});
+
+const calc = (x, y) => [-(y - window.innerHeight / 2) / 20, (x - window.innerWidth / 2) / 20, 1.1];
```

Comment:
`calc` doesn't explain much what this function does

Reply:
honestly, this function was took from a codepen demo of `react-spring` that didn't used spring's hooks, i'd just reimplement it in a hook component with modified behavior. But this function determine the position and scale of an item when hovered or dragged according to cursor position.